### PR TITLE
feat: allow custom seed in fractional op

### PR DESCRIPTION
**Pull request by colebaileygit**
## This PR

- updates fractional schema to accept complex rules as first argument to override targeting so that we can support using `cat` and other rules to provide custom seeds to randomisation

### Related Issues

https://github.com/open-feature/flagd/issues/1264
resolves https://github.com/open-feature/flagd/issues/1265

### How to test

see https://github.com/open-feature/flagd/pull/1266

### CODE REVIEW

File: json/targeting.yaml
Comment (beeme1mr) [inline json/targeting.yaml:392]: Nice, that's easy enough 👍

Comment (colebaileygit) [reply json/targeting.yaml:392]: Was skeptical at first since in theory some rules might not return strings, but I don't see a clever way to be so specific in the schema without greatly limiting flexibility

Comment (beeme1mr) [reply json/targeting.yaml:392]: @toddbaert any concerns? I believe `anyRule` should cover all valid operations.

Comment (beeme1mr) [reply json/targeting.yaml:392]: > Was skeptical at first since in theory some rules might not return strings, but I don't see a clever way to be so specific in the schema without greatly limiting flexibility

We should be casting the value to a string before using it in the hashing function.

**Review by beeme1mr — APPROVED** [review]
Approving but I would like to wait for @toddbaert to approve before we merge.

**Review by Kavindu-Dodan — APPROVED** [review]

**Pull request merged**
